### PR TITLE
[#56] Improve documentation for v0.2.0

### LINKED ISSUE #56: Improve documentation for v0.2.0
- [x] Generate new GIF
- [x] Add "Quick Start Guide"
- [x] Rewrite the beginning to specify that the tool is for downloading prebuilt binaries
- [x] Add limitations (no completion, man pages, files, only GitHub)
- [x] Add feature: cross-platform
- [x] Ensure all commands are mentioned in the README
- [x] Add the comparison with `cargo-binstall`: https://github.com/cargo-bins/cargo-binstall/
- [x] Add the comparison with `clyde`: https://github.com/agateau/clyde
- [x] Add the comparison with `topgrade`: https://github.com/r-darwish/topgrade
- [x] Update CHANGELOG
- [x] Bump up version
- [x] Write goals explicitly
- [x] Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #56

I've improved the documentation for `tool-sync` in a few ways:

* Rewrote README
* Added a CHAGELOG
* Slightly changed the CONTRIBUTING.md guide (now it requires to add new changes in the changelog)
* Generated a new GIF with the new output
* Increased the version to `0.2.0`

@MitchellBerend @FrancisMurillo @zixuanzhang-x please, double check the CHAGELOG.md file to make sure I haven't introduced any mistakes 🙂 

I want to implement #54 and #55 before actually creating a release so the CHANGELOG will change. But those changes shouldn't affect the documentation so this one can be merged separately.

### Additional tasks

- [x] Documentation for changes provided/changed
- [ ] Tests added

### CODE REVIEW

**Comment by chshersh** [comment]
Integration testing proved to be very useful!

image

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## [0.2.0] — 2022-09-?? 🔃
+
+### Added
+* [#57](https://github.com/chshersh/tool-sync/issues/57):
+  Adds the `install` command to install known tools
+  (by [@MitchellBerend][MitchellBerend])
+  **#CLI**
+* [#52](https://github.com/chshersh/tool-sync/issues/52),
+  [#73](https://github.com/chshersh/tool-sync/issues/73):
+  Add the `default-config` command to generate the default `.tools.toml` file
```
Comment (zixuanzhang-x) [inline CHANGELOG.md:20]:
```suggestion
  Add the `default-config` command to generate the default `.tool.toml` file
```

File: README.md [hunk 2]
```diff
@@ -14,54 +22,81 @@
...
 
 `tool-sync` has several distinguished features that allows you to manage your
 personal toolbox easily:
 
-* Installs the latest version of tools by default. You can easily update all
-  your tools with a single command!
-* Supports common tools that you can easily install without extra configuration
-* Automatically guesses asset name from your current OS
-* Configures via a simple TOML file
+* 🚀 Installs the latest version of tools by default
```
Comment (zixuanzhang-x) [inline README.md:48]: love those emojis!

**Review by zixuanzhang-x — APPROVED** [review]
LGTM! The README is very informative. Thanks for this documentation update!

File: README.md [hunk 3]
```diff
@@ -193,15 +249,15 @@ Run `tool --help` for more details.
...
+1. **Manual download**: you can download GitHub releases manually without using
    any extra tools.
 
    + **Pros**
      + No extra tools required, only your browser and unpack utility
    + **Cons**
      + Tedious manual process
 
-2. **GitHub CLI**. You can download assets from releases using
+2. **GitHub CLI**: You can download assets from releases using
```
Comment (FrancisMurillo) [inline README.md:260]: Just a minor question on capitalization consistency but overall looks good. The first word is capitalized in this numbered item but the others are lower case, so is this okay?

**Review by FrancisMurillo — APPROVED** [review]
Looks good.

**Review by MitchellBerend — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (chshersh) [reply README.md:260]: Definitely a typo and needs to be changed to be consistent with other list items 😌

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chshersh) [reply CHANGELOG.md:20]: This particular message was automatically generated from the PR, indeed. And @zixuanzhang-x noticed the typo (and been fixing this typo in the repository in many places)

**Pull request merged**